### PR TITLE
Fix popup position around marker that was moved to side globe

### PR DESCRIPTION
**Pull request by sbachinin**
Fixes https://github.com/maplibre/maplibre-gl-js/issues/2775

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [ ] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Add an entry to `CHANGELOG.md` under the `## main` section.

Idea is to set lngLat every time the popup is opened (when Popup.addTo is called) and to reset the previous position.
This prevents smartWrap from doing its bad job.

https://github.com/maplibre/maplibre-gl-js/assets/11789697/fe6eadd0-1931-40f2-ae07-7e537513bca5

### CODE REVIEW

File: src/ui/marker.ts [hunk 1]
```diff
@@ -518,7 +517,9 @@ export class Marker extends Evented {
 
         if (!popup) return this;
         else if (popup.isOpen()) popup.remove();
-        else popup.addTo(this._map);
+        else {
+            popup.setLngLat(this._lngLat); popup.addTo(this._map);
```
Comment (HarelM) [inline src/ui/marker.ts:521]: Will this work with a popup the uses `trackPointer`, or am I mixing stuff that are not related?

File: src/ui/marker.test.ts [hunk 2]
```diff
@@ -6,11 +6,11 @@ import Point from '@mapbox/point-geometry';
 import simulate from '../../test/unit/lib/simulate_interaction';
 import type {Terrain} from '../render/terrain';
 
-function createMap(options = {}) {
+function createMap(options = {} as any) {
```
Comment (HarelM) [inline src/ui/marker.test.ts:9]: Can this be typed better?

File: src/ui/marker.ts (hunk 1 above)
Comment (HarelM) [reply src/ui/marker.ts:521]: Also, please split this into two lines or chain the calls (or both).

**Comment by HarelM** [comment]
Added a few nit-picking, overall looks good.

File: src/ui/marker.ts (hunk 1 above)
Comment (sbachinin) [reply src/ui/marker.ts]: As for trackPointer, this pr changes the current behaviour but for the better.
What's happening on main if you call `trackPointer()` on a popup that's attached to marker:
https://github.com/maplibre/maplibre-gl-js/assets/11789697/a77e7433-9537-4cdd-9a16-a20e6e467301
I think the proper behaviour is to ignore such trackPointer call.

After this change `_trackPointer` prop will be set to false whenever popup is opened (it's set to false by `Popup.setLngLat`). Thus popup will just stay with a "parent" marker. It look like a proper behaviour to me.

Comment (HarelM) [reply src/ui/marker.ts]: Ok, thanks.

**Comment by HarelM** [comment]
Please sdd a change log and I'll merge this.

**Review by HarelM — APPROVED** [review]

**Comment by sbachinin** [comment]
Sure, I'd be happy to help with reviews etc. Please add me.
